src/Data/Lens/Prism.purs

```diff
@@ -18,23 +57,78 @@ import Data.Profunctor (dimap, rmap)
 import Data.Profunctor.Choice (right)
 import Data.Newtype (under)
 
--- | Create a `Prism` from a constructor/pattern pair.
+-- | Create a `Prism` from a constructor and a "focus" function that
+-- | produces an `Either`:
+-- | 
+-- | ```purescript
+-- | solidFocus :: Prism' Fill Color
+-- | solidFocus = prism Solid case _ of
+-- |   Solid color -> Right color
+-- |   anotherCase -> Left anotherCase
+-- | ```
 prism :: forall s t a b. (b -> t) -> (s -> Either t a) -> Prism s t a b
 prism to fro pab = dimap fro (either id id) (right (rmap to pab))
 
+-- | Create a `Prism` from a constructor and a "focus" function that
+-- | produces a `Maybe`:
+-- | 
+-- | ```purescript
+-- | solidFocus :: Prism' Fill Color
+-- | solidFocus = prism' Solid case _ of
+-- |   Solid color -> Just color
+-- |   _ -> Nothing
+-- | ```
 prism' :: forall s a. (a -> s) -> (s -> Maybe a) -> Prism' s a
 prism' to fro = prism to (\s -> maybe (Left s) Right (fro s))
 
--- | Review a value through a `Prism`.
-review :: forall s t a b. Review s t a b -> b -> t
-review = under Tagged
+-- | Create a prism that focuses on only some of the values of a case,
```

Comment:
In order for this to work as a law-abiding prism, the predicate should identify only the value passed it. It's identifying the one value as a summand of the whole.

Reply:
I don't understand "identify only the value passed it".

Reply:
I'll make explicit `only`'s dependence on `Eq`.

Reply:
What would be a use of `nearly` that would make an ill-behaved prism?

Reply:
> I don't understand "identify only the value passed it".

I mean that `f a` can only hold when `a = x` and not otherwise.

> I'll make explicit only's dependence on Eq.

It's not just the dependence on `Eq`, but that that instance must be definitional equality: `eq x y` if and only if `x = y` definitionally.

> What would be a use of nearly that would make an ill-behaved prism?

Your example gives a prism which does not obey the second prism law. Extracted from the `lens` docs:

> If `preview l s ≡ Just a` then `review l a ≡ s`

i.e. if `predicate s` then `const (Solid referenceColor) = s`. This obviously only holds for one value of `s`, not all of them.

Reply:
I completely misunderstood the purpose of `nearly` if there's only one possible `s` that can satisfy the predicate. Is it then the case that the difference between `nearly` and `only` just that the latter uses `Eq` whereas the former uses the given predicate to detect the one-and-only-one value that's allowed?

Reply:
That's right. Per the `lens` docs:

> To comply with the Prism laws the arguments you supply to nearly a p are somewhat constrained.
> 
> We assume p x holds iff x ≡ a. Under that assumption then this is a valid Prism.